AprilSylph/XKit-Rewritten
# Themed Posts: bugfix for deleted blogs, new option to control deleted blog theming

### Description

Fixes #874, and introduces a new option to control the theming behavior for deleted/blacklisted posts (described in #874 under "expected behavior").

### Testing steps

1. import these settings
    ```json
    {
      "enabledScripts": ["themed_posts"],
      "themed_posts.preferences.blacklistedUsernames": "",
      "themed_posts.preferences.enableOnPeepr": false,
      "themed_posts.preferences.missingPostMode": "palette",
      "themed_posts.preferences.reblogTrailTheming": true
    }
    ```
1. either:
   - a) scroll down to a post containing a deactivated blog
   - or b) scroll down to a post containing multiple blogs, and add one user somewhere in the middle of the trail to the themed posts blacklist
1. set your palette to something easily distinguishable from any of the themes in that post (in my screenshots I used the built in `Low-Contrast Classic`)
1. the deleted/manually blacklisted blog should have the colors of the palette.
   ![image](https://user-images.githubusercontent.com/16143061/204404310-23001106-6c0b-4a94-8e98-2b7c5e5561ce.png)
1. change `Themed Posts`->`Disabled/deleted blog theming` to `inherit reblogging blog's theme`
1. the deleted/manually blacklisted blog should now have the colors of the reblogging user's theme
   ![image](https://user-images.githubusercontent.com/16143061/204404635-ca3aca91-9c77-4b6a-869a-1c67914e0a12.png)
